core-services/src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -73,4 +63,20 @@ public String calculatePartition(final EventType eventType, final JSONObject eve
         }
     }
 
+    @Override
+    public String calculatePartition(
+            final JSONObject event, final List<String> eventKeys, final List<String> partitions) {
+
+        final int hashValue = eventKeys.stream()
```

Comment:
this can replaced with 

```
final int hashValue = eventKeys.stream()
            .mapToInt(s -> stringHash.hashCode(s))
            .sum();
```